### PR TITLE
enhance: Optimeize clustering compaction

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #30633

### CODE REVIEW

File: pkg/util/paramtable/component_param.go [hunk 1]
```diff
@@ -4081,7 +4081,7 @@ if this parameter <= 0, will set it as 10`,
 	p.ClusteringCompactionMemoryBufferRatio.Init(base.mgr)
 
 	p.ClusteringCompactionWorkerPoolSize = ParamItem{
-		Key:          "datanode.clusteringCompaction.cpu",
+		Key:          "dataNode.clusteringCompaction.cpu",
```
Comment (wayblink) [inline pkg/util/paramtable/component_param.go:4084]: how about change cpu -> workPoolSize

File: pkg/util/paramtable/component_param.go [hunk 2]
```diff
@@ -4071,7 +4071,7 @@ if this parameter <= 0, will set it as 10`,
 	p.SlotCap.Init(base.mgr)
 
 	p.ClusteringCompactionMemoryBufferRatio = ParamItem{
-		Key:          "datanode.clusteringCompaction.memoryBufferRatio",
+		Key:          "dataNode.clusteringCompaction.memoryBufferRatio",
 		Version:      "2.4.2",
```
Comment (wayblink) [inline pkg/util/paramtable/component_param.go:4075]: update 2.4.2->2.4.6

File: internal/datanode/compaction/clustering_compactor.go [hunk 3]
```diff
@@ -593,17 +645,14 @@ func (t *clusteringCompactionTask) writeToBuffer(ctx context.Context, clusterBuf
 	defer t.clusterBufferLocks.Unlock(clusterBuffer.id)
 	// prepare
 	if clusterBuffer.writer == nil {
-		err := t.refreshBufferWriter(clusterBuffer)
-		if err != nil {
-			return err
-		}
+		log.Warn("unexpected behavior, please check", zap.Int("buffer id", clusterBuffer.id))
+		panic(fmt.Sprintf("unexpected behavior, please check buffer id: %d", clusterBuffer.id))
```
Comment (wayblink) [inline internal/datanode/compaction/clustering_compactor.go:649]: How about return error, fail the compaction task? panic is too heavy

**Comment by xiaocai2333** [comment]
rerun ut

**Pull request merged**